DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/CourseSetupControllerTests.cs

```diff
@@ -151,22 +219,446 @@ public void Index_with_CLEAR_filterBy_and_new_filter_query_parameter_sets_new_co
+            var applicationOptionsSelectList = new List<SelectListItem> { new SelectListItem("Test Name", "1") };
+            var model = new SelectCourseViewModel(application.ApplicationId, applicationOptionsSelectList);
+            SetAddNewCentreCourseTempData();
+
+            // When
+            var result = controller.SelectCourse(model);
+
+            // Then
+            controller.TempData.Peek<AddNewCentreCourseData>()!.Application.Should()
+                .BeEquivalentTo(application);
+            controller.TempData.Peek<AddNewCentreCourseData>()!.SetCourseDetailsModel.Should().BeNull();
+            controller.TempData.Peek<AddNewCentreCourseData>()!.SetCourseOptionsModel.Should().BeNull();
+            controller.TempData.Peek<AddNewCentreCourseData>()!.SetCourseContentModel.Should().BeNull();
+            controller.TempData.Peek<AddNewCentreCourseData>()!.SetSectionContentModels.Should().BeNull();
+            result.Should().BeRedirectToActionResult().WithActionName("SetCourseDetails");
+        }
+
+        [Test]
+        public void SelectCourse_does_not_redirect_with_invalid_model()
+        {
+            var model = new SelectCourseViewModel { ApplicationId = 1 };
+            controller.ModelState.AddModelError("ApplicationId", "Select a course");
+            SetAddNewCentreCourseTempData(application);
+
+            // When
+            var result = controller.SelectCourse(model);
+
+            // Then
+            result.Should().BeViewResult().ModelAs<SelectCourseViewModel>();
+            controller.ModelState["ApplicationId"].Errors[0].ErrorMessage.Should()
+                .BeEquivalentTo("Select a course");
+        }
+
+        [Test]
+        public void
+            SaveCourseDetails_correctly_adds_model_error_if_customisation_name_is_not_unique()
         {
             // Given
-            var controllerWithNoCookies = new CourseSetupController(
-                    courseService,
-                    courseCategoryDataService,
-                    courseTopicsDataService
+            var formData = GetSetCourseDetailsViewModel();
+
+            A.CallTo(
+                () => courseService.DoesCourseNameExistAtCentre(
+                    "Name",
+                    101,
+                    1,
+                    0
                 )
-                .WithDefaultContext()
-                .WithMockUser(true);
+            ).Returns(true);
 
             // When
-            var result = controllerWithNoCookies.Index();
+            var result = controller.SetCourseDetails(formData);
 
             // Then
-            result.As<ViewResult>().Model.As<CourseSetupViewModel>().FilterBy.Should()
-                .Be("Status|Active|true");
+            result.Should().BeViewResult().ModelAs<SetCourseDetailsViewModel>();
+            controller.ModelState["CustomisationName"].Errors[0].ErrorMessage.Should()
+                .BeEquivalentTo("Course name must be unique, including any additions");

```

Comment:
Can use `Be(..)` here instead of `.BeEquivalentTo(..)`

Reply:
done